deployment/SampleClaimsProvider/ClaimsProviderStore.cs

```diff
@@ -0,0 +1,468 @@
+
+                    bool matchAbsolute = !path.EndsWith("*", StringComparison.OrdinalIgnoreCase);
+
+                    foreach (StringClaim claim in claims)
+                    {
+                        if (claim.Name.StartsWith("//", StringComparison.OrdinalIgnoreCase))
+                        {
+                            if (matchAbsolute && claim.Name.Equals(pathPrefix, StringComparison.OrdinalIgnoreCase))
+                            {
+                                filteredClaims.Add(claim);
+                            }
+                            else if (!matchAbsolute && claim.Name.StartsWith(pathPrefix, StringComparison.OrdinalIgnoreCase))
+                            {
+                                filteredClaims.Add(claim);
+                            }
+                        }
+                        else
+                        {
+                            if (path.StartsWith(McvpPathPrefix, StringComparison.OrdinalIgnoreCase))
+                            {
+                                string trimmedPathPrefix = pathPrefix.Substring(McvpPathPrefix.Length);
+
+                                if (matchAbsolute && claim.Name.Equals(trimmedPathPrefix, StringComparison.OrdinalIgnoreCase))
+                                {
+                                    filteredClaims.Add(claim);
+                                }
+                                else if (!matchAbsolute && claim.Name.StartsWith(trimmedPathPrefix, StringComparison.OrdinalIgnoreCase))
+                                {
+                                    filteredClaims.Add(claim);
+                                }
+                            }
+                        }
+                    }
+                }
+
+                filteredLabelClaims[labelsAndPath.Key] = filteredClaims.ToList();
+            }
+
+            return filteredLabelClaims;
+        }
+
+        private async Task<List<StringClaim>> LookupClaimsByEntityId(string entityId)
+        {
+            string documentId = DocumentIdCreator.CreateEntityClaimDocumentId(entityId);
+            string partitionKey = DocumentIdCreator.CreateEntityPartition(entityId);
+
+            Uri documentUri = UriFactory.CreateDocumentUri(this.database.Id, this.collection.Id, documentId);
+            DocumentResponse<ClaimsDocument> entityRecordResponse = await this.documentClient.ReadDocumentAsync<ClaimsDocument>(documentUri, new RequestOptions { PartitionKey = new PartitionKey(partitionKey) });
+            ClaimsDocument entityRecord = entityRecordResponse.Document;
+            if (entityRecord != null)
+            {
+                return entityRecord.Claims;
+            }
+            else
+            {
+                return null;
+            }
+        }
+
+        private IDictionary<string, IList<StringClaim>> LookupClaimsByAuthToken(string authToken, IDictionary<string, IList<string>> labelsAndPaths)
```

Comment:
If we're going to leave this as a TODO, can we provide the customer more information about what this implementation might be? What does it mean to lookup claims by auth token? How does the auth token have claims? Do I need to validate this token? 

As is this doesn't help the customer understand the value of this.

Reply:
This has been open a long time without resolution.